objects/client.c

```diff
@@ -363,151 +377,219 @@
+ * <tr><td><b>menu</b></td><td>A context menu.</td></tr>
+ * <tr><td><b>toolbar</b></td><td>A floating toolbar.</td></tr>
+ * <tr><td><b>utility</b></td><td></td></tr>
+ * <tr><td><b>dropdown_menu</b></td><td>A context menu attached to a parent position.</td></tr>
+ * <tr><td><b>popup_menu</b></td><td>A context menu.</td></tr>
+ * <tr><td><b>notification</b></td><td>A notification popup.</td></tr>
+ * <tr><td><b>combo</b></td><td>A combobox list menu.</td></tr>
+ * <tr><td><b>dnd</b></td><td>A drag and drop indicator.</td></tr>
+ * <tr><td><b>normal</b></td><td>A normal application main window.</td></tr>
+ * </table>
  *
  * More information can be found [here](https://specifications.freedesktop.org/wm-spec/wm-spec-latest.html#idm140200472629520)
  *
- * **Signal:**
- *
- *  * *property::type*
- *
  * @property type
- * @param string
+ * @tparam string type
+ * @propemits false false
+ * @see ruled.client
  */
 
 /**
  * The client class.
  *
- * To get a client class from the command line, use the command `xprop
- * WM_CLASS`. The class will be the second string.
+ * A class usually maps to the application name. It is useful in
+ * the rules to apply different properties to different clients. It
+ * is also useful, along with `instance`, so implement "windows counter"
+ * used in many popular docks and Alt-Tab like popups.
+ *
+ * To get a client class from the command line, use the command:
  *
- * **Signal:**
+ *    xprop WM_CLASS
  *
- *  * *property::class*
+ * The class will be the second string.
+ *
+ * This *should* never change after the client is created, but some
+ * buggy application like the Spotify desktop client are known to
+ * violate the specification and do it anyway. There *is* a signal for
+ * this property, but it should hopefully never be useful. If your
+ * applications change their classes, please report a bug to them.
+ * It tends to break `ruled.client` and other AwesomeWM APIs.
  *
  * @property class
- * @param string
+ * @tparam string class
+ * @propemits false false
+ * @see instance
+ * @see ruled.client
  */
 
 /**
  * The client instance.
  *
- * To get a client instance from the command line, use the command `xprop
- * WM_CLASS`. The instance will be the first string.
+ * The `instance` is a subtype of the `class`. Each `class` can have
+ * multiple instances. This is useful in the `ruled.client` rules to
+ * filter clients and apply different properties to them.
  *
- * **Signal:**
+ * To get a client instance from the command line, use the command:
  *
- *  * *property::instance*
+ *     xprop WM_CLASS
+ *
+ * The instance will be the first string.
  *
  * @property instance
- * @param string
+ * @tparam string instance
+ * @propemits false false
+ * @see class
+ * @see ruled.client
  */
 
 /**
  * The client PID, if available.
  *
- * **Signal:**
- *
- *  * *property::pid*
+ * Some clients might have funky PID `fork` behavior, but
+ * normally this will never change.
```

Comment:
No idea what this is supposed to mean

Reply:
I replaced the text. tl;dr; I referred to `fork()` and how it can mess things up. As far as I understand (please correct me), it is even possible kill a process and have another one later on claim ownership of a window. Even as far as transferring ownership to another computer. Do the *protocol* explicitly prevents this? I am aware most (all) GUI toolkits explicitly prevent this.

Reply:
When a process dies, its X11 connection is closed. The X11 server will then destroy all resources that are owned by that client, thus the window should be closed. One can tell the X11 server "please don't destroy my resources when I disconnect", but the only use that I ever saw of that was the wallpaper (you do not want the wallpaper to disappear when the wallpaper setter program closes its X11 connection).

There is no built-in way to transfer ownership, I think, but "ownership" is just "which stuff has to be destroyed when this X11 connection closes", nothing else (AFAIK).

Reply:
Ok, thanks for the clarification. That demo where someone was moving an application across computers was probably using some connections proxies or something, I can't even find it anymore. I removed that paragraph.